packages/experimental/Shadow/src/Shadow.types.ts

```diff
@@ -0,0 +1,22 @@
+import { IBorderTokens } from '@fluentui-react-native/tokens';
+import { ViewStyle, ViewProps, ColorValue } from 'react-native';
+
+export type ShadowDepth = '2' | '4' | '8' | '16' | '28' | '64';
+
+export interface ShadowTokens extends IBorderTokens {
+  ambient?: ViewStyle;
+  key?: ViewStyle;
+  color?: ColorValue;
+  backgroundColor?: ColorValue;
+  innerShadowSlot?: React.ComponentType<ViewProps>;
+  depth2?: ShadowTokens;
+  depth4?: ShadowTokens;
+  depth8?: ShadowTokens;
+  depth16?: ShadowTokens;
+  depth28?: ShadowTokens;
+  depth64?: ShadowTokens;
+}
+
+export interface ShadowProps extends ViewProps, ViewStyle, IBorderTokens {
```

Comment:
I'm curious why this needs to extend ViewStyle and BorderTokens? For changing tokens I would suggest encouraging people to use .customize(). For style changes (e.g. ViewStyle) the internal memoization in mergeStyles should handle the overriding by setting the style prop. In other words it feels like this component can start as a View that has an additional depth prop. It's always easy to add more later.

Reply:
I didn't love this and its the part of the reason I wanted to start soliciting feedback because it worked for my hackathon needs but didn't feel right.

I guess my thought was if we don't expose ViewStyle props as props on the Shadow we get in a weird place for a lot of users.  A standalone Shadow component is a fairly basic concept kind of like a View++, and it's hard for me to imagine people not tweaking things like borderRadius and padding on it regularly.  From an outsider's perspective, I want to treat it like it's a single View.  Internally, I'm merging the user provided style prop onto the root slot, but borderRadius is needed on both slots, padding needs to go on inner slot, as well as I presume many of the flexbox style props.

Leaving ViewStyle out of the props interface means that anyone that needs something to go on the inner slot customize it.  Maybe that's ok?

Reply:
That makes more sense now. I do think the default should be to start with .customize only and then add things to props as needed as it is much harder to go the other way. That being said if we think there are a set of commonly modified values then it might be worth exposing them as props.

Maybe a way to reason about it is if you are rendering a page and you want to change how all of your shadows look customize works great. If you want them to be much more dynamic and may have lots of permutations then props would work.

Right now we expect people to have multiple depth values on a page which is why its a prop. (I'm just trying to work through how to make these decisions)